src/Workleap.AspNetCore.Authentication.ClientCredentialsGrant/OpenAPI/SwaggerUtils.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Workleap.AspNetCore.Authentication.ClientCredentialsGrant.OpenAPI;
+
+internal static class SwaggerUtils
+{
+    public static IEnumerable<string> GetRequiredPermissions(ApiDescription apiDescription)
+    {
+        List<RequireClientCredentialsAttribute> attributes = new();
+
+        if (apiDescription.TryGetMethodInfo(out var methodInfo))
+        {
+            // Controllers - Attributes on the action method (empty for minimal APIs)
+            var methodAttributes = methodInfo.GetCustomAttributes<RequireClientCredentialsAttribute>(inherit: true).ToList();
+
+            if (methodAttributes.Count != 0)
+            {
+                attributes.AddRange(methodAttributes);
+            }
+            else if (methodInfo.DeclaringType != null)
+            {
+                // Controllers - Attributes on the controller class itself if was not overridden by the action method
+                attributes.AddRange(methodInfo.DeclaringType.GetCustomAttributes<RequireClientCredentialsAttribute>(inherit: true));
```

Comment:
We should also validate it work correctly when set at the class level, I am not sure